### PR TITLE
Move SearchIcon to icon folder and create story

### LINKED ISSUE #13563: Move SearchIcon to icon folder and create story
**Description**
Move `SearchIcon` component file from `ui/components/ui/search-icon/search-icon.component.js` to `ui/components/ui/icon/search-icon.component.js`

Add to icon stories `ui/components/ui/icon/icon.stories.js` under "Other Icons"

Screen Shot 2022-02-09 at 10 37 15 AM

### PR DESCRIPTION
**Pull request by rockingrohit9639**
Fixes: #13563 

image

### CODE REVIEW

File: ui/components/ui/icon/search-icon.component.js
Comment (georgewrmarshall) [inline ui/components/ui/icon/search-icon.component.js:17]: I think this can be simplified a bit

```suggestion
    <svg
      className={className}
      height={size}
      width={size}
      xmlns="http://www.w3.org/2000/svg"
      fill="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        fill={color}
        fillRule="evenodd"
        clipRule="evenodd"
        d="M10.9977 19.9955C13.0486 19.9955 15.0272 19.2953 16.6141 18.0274L20.2882 21.7015C20.3807 21.7955 20.4909 21.8703 20.6125 21.9215C20.7341 21.9728 20.8646 21.9994 20.9965 22C21.1285 22.0005 21.2592 21.9749 21.3812 21.9247C21.5032 21.8745 21.614 21.8006 21.7073 21.7073C21.8006 21.614 21.8745 21.5032 21.9247 21.3812C21.9749 21.2592 22.0005 21.1285 22 20.9965C21.9994 20.8646 21.9728 20.7341 21.9215 20.6125C21.8703 20.4909 21.7955 20.3807 21.7014 20.2882L18.0274 16.6141C19.2953 15.0272 19.9955 13.0486 19.9955 10.9977C19.9955 9.81614 19.7627 8.64611 19.3106 7.55445C18.8584 6.4628 18.1956 5.47089 17.3601 4.63538C16.5246 3.79986 15.5327 3.13709 14.441 2.68491C13.3494 2.23273 12.1793 2 10.9977 2C9.81614 2 8.64611 2.23273 7.55445 2.68491C6.4628 3.13709 5.47089 3.79986 4.63538 4.63538C3.79986 5.47089 3.13709 6.4628 2.68491 7.55445C2.23273 8.64611 2 9.81614 2 10.9977C2 13.3841 2.94797 15.6727 4.63538 17.3601C6.32278 19.0475 8.61139 19.9955 10.9977 19.9955ZM11 18C14.866 18 18 14.866 18 11C18 7.13401 14.866 4 11 4C7.13401 4 4 7.13401 4 11C4 14.866 7.13401 18 11 18Z"
      />
    </svg>
```

This should also fix the size issues

**Before**

Screen Shot 2022-02-22 at 8 37 31 PM

**After**

Screen Shot 2022-02-22 at 8 38 00 PM

File: ui/components/ui/icon/search-icon.component.js
Comment (georgewrmarshall) [inline ui/components/ui/icon/search-icon.component.js:23]: Remove this in favour of default parameters. `defaultProps` will eventually be deprecated.

```suggestion
```

File: ui/components/ui/icon/search-icon.component.js
Comment (georgewrmarshall) [inline ui/components/ui/icon/search-icon.component.js:4]: Update to use [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

```suggestion
export default function SearchIcon({ className = undefined, size, color }) {
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Thanks for picking this up! We are just waiting on #13214 to be merged before we can merge this as it will need to be updated in those instances. Have made a couple suggestions.

**Comment by mcmire** [comment]
Hi @rockingrohit9639 it looks like there are some extra changes showing up in this PR, likely due to things being out of sync. Would you mind rebasing this? If you are encountering merge conflicts then let us know and we can help.

**Comment by georgewrmarshall** [comment]
Closing as no updates in a while. Let me know if you would like to give this another go @rockingrohit9639. Thanks!

**Pull request closed without merging**
